### PR TITLE
feat(console): display user password information on user details page

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

This PR introduces a new feature for managing user passwords within the user settings page of the console.

Key changes include:

- Components:
   - **UserPassword**: A new component that displays the user's password status and provides options to reset or generate a password. It integrates a modal for the password reset form and updates the user interface accordingly.
   - **ResetPasswordForm**: Adjusted to handle both resetting and generating passwords, with appropriate title changes based on the user's current password status.
- Localization: 
   - Updated localization files to include new keys for password management, ensuring the interface is accessible in multiple languages.
- Refactoring: 
   - Removed legacy code related to password resetting from the `UserDetails` component, streamlining the codebase.

## Testing

image
image
image
image
image
image

## Checklist

- [x] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: .changeset/polite-bats-learn.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@logto/core": patch
```
Comment (wangsijie) [inline .changeset/polite-bats-learn.md:2]: minor?

Comment (xiaoyijun) [reply .changeset/polite-bats-learn.md:2]: This pr needs to rebase.

**Comment by xiaoyijun** [comment]
Waitting rebase.

**Comment by charIeszhao** [comment]
The `Password already set` might need to be improved? We use asterisks in profile page
image

**Review by wangsijie — APPROVED** [review]

**Comment by xiaoyijun** [comment]
> The `Password already set` might need to be improved? We use asterisks in profile page image

@charIeszhao 

I previously synced with the designer about this design. Since Logto doesn’t store user passwords, we don’t know how many characters the password contains. Displaying asterisks was meant to hint at the password length for the user. Perhaps these asterisks should be replaced with ‘Password already set.’

**Comment by xiaoyijun** [comment]
> > The `Password already set` might need to be improved? We use asterisks in profile page image
> 
> @charIeszhao
> 
> I previously synced with the designer about this design. Since Logto doesn’t store user passwords, we don’t know how many characters the password contains. Displaying asterisks was meant to hint at the password length for the user. Perhaps these asterisks should be replaced with ‘Password already set.’

Per offline discussion, keep the stauts quo for admin console, and the profile page will have a new render logic for the password field.

**Review by gao-sun — APPROVED** [review]
lgtm on changeset

**Pull request merged**